File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        vector<int> v;

        ListNode* current = head;
        while (current != nullptr) {
            v.push_back(current->val);
            current = current->next;
        }

        ListNode* middleNode = head;
        for (int i = 0; i < v.size()/2; ++i) {
            middleNode = middleNode->next;
        }

        return middleNode;
    }
};

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
import java.util.Vector;

class Solution {
    public ListNode middleNode(ListNode head) {
        Vector<Integer> v = new Vector<>();

        ListNode current = head;
        while (current != null) {
            v.add(current.val);
            current = current.next;
        }

        ListNode middleNode = head;
        for (int i = 0; i < v.size()/2; ++i) {
            middleNode = middleNode.next;
        }
        return middleNode;
    }
}